scripts/sanity_chk/sanitylib.py

```diff
@@ -3297,6 +3299,57 @@ def csv_report(self, filename):
                     rowdict["rom_size"] = rom_size
                 cw.writerow(rowdict)
 
+    def json_report(self, filename):
+        rowdict = {}
+        rowdict["environment"] = {"os": os.name,
+                                    "toolchain": self.get_toolchain()
+                                    }
+        rowdict["discards"] = []
+        for instance, reason in self.discards.items():
+            rowdict["discards"].append({"test": instance.testcase.name,
+                        "arch": instance.platform.arch,
+                        "platform": instance.platform.name,
+                        "reason": reason})
+        rowdict["testspec"] = []
+        for instance in self.instances.values():
+            rowdict["testspec"].append({"test": instance.testcase.name,
+                                        "arch": instance.platform.arch,
+                                        "platform": instance.platform.name,
+                                        "handler": instance.platform.simulation,
+                                        "status": instance.status,
+                                        "extra_args": instance.testcase.extra_args,
+                                        "extra_configs": instance.testcase.extra_configs,
+                                        "harness": instance.testcase.harness,
+                                        "harness_config": instance.testcase.harness_config,
+                                        "type": instance.testcase.type,
+                                        "tags": list(instance.testcase.tags),
+                                        "build only": instance.testcase.build_only
+                                            })
+            if instance.status in ["passed"]:
+                if instance.handler:
+                    rowdict["testspec"].append({"handler_time": (instance.metrics.get           ("handler_time", 0))})
```

Comment:
you are now adding handler_time as a new list entry of the testspec which is not what we want, this should be in the same dict you started above.